chichiwang/exercise-js
# Separate webpack-config into modules

## Description
Linked to Issue: #17
Separate Webpack configurations out into multiple modules for easier composability and maintenance. In the future it will be easier to separate out different configurations for different target environments this way (development, production, etc).

## Changes
* Create a directory `webpack-config/` to house config modules for webpack
  * Create a file `webpack-config/paths.js` to house all of the directory and file paths used in webpack configurations
  * Create a file `webpack-config/base.js` to house base configs used in both client and server configurations
  * Create a file `webpack-config/client.js` to house client-specific webpack configurations
  * Create a file `webpack-config/server.js` to house server-specific webpack configurations
* Update `webpack-config.js` to import the new modules `webpack-config/server.js` and `webpack-config/client.js`

## Steps to QA
* Pull down and switch to this branch
* Run `npm run build:dev` and ensure no errors, as well as the expected artifacts to have been generated
  * Run `node dist/index.js` to ensure no errors, server starts. Verify functionality in browser.
* Run `npm start` to ensure watch process starts without issue.
  * Verify functionality in browser.
  * Verify watcher properly rebuilds when changing non-test files in `app/`, `client/`, and `server/`
